### ISSUE TITLE
bb-modal: dialog element missing a programmatic label (2.4.6)

### ISSUE THREAD

**Issue by Blackbaud-MattGregg** [blackbaud/skyux1#898]
### Expected behavior
Dialog element has a text label. Ideally this is existing text in the UI (i.e., aria-labelledby property of the role="dialog" element points to the id of the text element)
https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal

### Actual behavior
Dialog element doesn't have a label
